socs/agents/tank_level_monitor/agent.py

```diff
@@ -0,0 +1,340 @@
+import argparse
+import os
+import struct
+import time
+
+import serial
+import txaio
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import Pacemaker, TimeoutLock
+
+
+class TankLevelMonitor:
+
+    """Initialize serial communication with tank sensors & creates dicts to store data depending on tanks numbers."""
+
+    def __init__(self):
+
+        self.client = serial.Serial("/dev/ttyUSB1", 9600, timeout=0.5)
+
+        self.tank_data = {}  # Define dictionary for Tank data, 7 fields
+        self.tank1_data = {}
+        self.tank2_data = {}
+        self.tank1_data_fields = ['Tank1_vol', 'Net1_vol', 'emp1_vol', 'prod1_h', 'water1_h', 'avg1_temp', 'water1_vol']
+        self.tank2_data_fields = ['Tank2_vol', 'Net2_vol', 'emp2_vol', 'prod2_h', 'water2_h', 'avg2_temp', 'water2_vol']
+
+        self.tank_data_ok = False
+        self.tank_length_to_read = 89
+        self.verbose = True
```

Comment:
A verbosity flag works for debugging, but I'll make the same optional suggestion as I made in #640 -- an alternative approach would be to use debug logging. This would allow you to drop the tracking of `verbose`/`verbosity` and instead use:

```
self.log.debug('debug statement')
```

The advantage here is that the `LOGLEVEL` environment variable can be changed without modification of this `verbosity` constant in the code.

If you want to implement this check out the [Logging documentation](https://ocs.readthedocs.io/en/main/developer/agent_references/logging.html#). (@felipecarrero also just implemented this, you can ask him as well.) I wouldn't let it hold up the PR though.